### ISSUE TITLE
Fragments extending from AndroidScopeComponent can't fetch dependencies before being attached

### ISSUE THREAD

**Issue by ZackOPP** [InsertKoinIO/koin#1580]
**Describe the bug**
When using a fragment that extends from `AndroidScopeComponent`, the fragment is expected to be attached to an activity before trying to fetch any dependency via Koin, meaning that we are at least expected to fetch the dependencies once the fragment is attached.

**To Reproduce**
Create a fragment as follows:
```kotlin
class MyFragment : Fragment(), AndroidScopeComponent {
    
    override val scope: Scope by fragmentScope()

    // This works as long as it's called after the fragment is attached to an activity
    private val myLazyDependency by inject<Foo>()

    // This crashes since Koin is now forced to link fragment's scope to activity's scope    
    private val nonLazyDependency = get<Foo2>() 
}
```

**Expected behavior**
Be able to fetch dependencies before the fragment is attached when it extends from `AndroidScopeComponent`

**Koin project used and used version (please complete the following information):**
`koin-android version 3.4.0`

**Comment by arnaudgiuliani** [comment]
This extra param can help you inject into a Fragment scope, without forcing to have the parent activity scope.

![image](https://github.com/InsertKoinIO/koin/assets/2360688/de01afa8-3040-44ec-9b2d-5dfdddb7f3b3)

**Comment by arnaudgiuliani** [comment]
fixed in `koin-android:3.4.1`
